specifications/transactions/onchain/README.md

```diff
@@ -0,0 +1,107 @@
+
+## Peer to Peer Payments and Transaction Metadata
+
+Most transactions will use the [peer_to_peer_with_metadata transaction script](https://github.com/diem/diem/blob/master/language/diem-framework/transaction_scripts/peer_to_peer_with_metadata.move). This single transaction represents all current transfers between two participants and distinguishes the types of transfers via the embedded metadata.
+
+The metadata is represented by the following `Rust` enum encoded in [Binary Canonical Serialization (BCS)](https://github.com/diem/bcs):
+
+```
+enum Metadata {
+  Undefined,
+  GeneralMetadata(GeneralMetadata),
+  TravelRuleMetadata(TravelRuleMetadata),
+  UnstructuredByteMetadata(Option<Vec<u8>>),
+  RefundMetadata(RefundMetadata),
+  CoinTradeMetadata(CoinTradeMetadata),
+}
+```
+
+Note: This is the canonical list and should be referred to in future DIPs so that authors need not reproduce the list in future DIPs.
+
+## Payments Using GeneralMetadata
+
+```
+enum GeneralMetadata {
+   GeneralMetadataV0(GeneralMetadatav0),
+}
+
+struct GeneralMetadataV0 {
+   Option<Vec<u8>> to_subaddress,
+   Option<Vec<u8>> from_subaddress,
+   Option<u64> referenced_event, // Deprecated
+}
+```
+
+GeneralMetadata leverages the notion of subaddresses to indicate a source and destination and are stored in the fields `from_subaddress` and `to_subaddress`, respectively.
+
+Subaddresses have the following properties:
+* 8-bytes
+* Subaddresses should be unique
+* The address represented by 8 zero bytes (or None/Null within the GeneralMetadataV0) is reserved to denote the root (VASP owned) account
+
+Lifetime of subaddresses:
+* There are no explicit lifetimes of subaddresses
+* The same `from_address` may receive multiple payments from distinct `to_subaddress`
+* A `from_subaddress` may be the recipient or a `to_subaddress` in an ensuing transaction
+* `to_subaddress` should be generated fresh each time upon user request
+* `from_subaddress` should be unique for each transaction
+
+Subaddresses should be used with great care to not accidentally leak personally identifiable information (PII). However, implementors must be mindful of the permissive nature of subaddresses as outlined in this specification.
+
+## Payments Using TravelRuleMetadata
+
+```
+enum TravelRuleMetadata {
+   TravelRuleMetadataVersionV0(OffChainReferenceId),
+}
+
+type OffChainReferenceId = Option<String>;
+```
+
+The TravelRuleMetadata completes a transaction that began with a pre-flight, or off-chain exchange. During this exchange, the two parties should have agreed on an `OffChainReferenceId`, likley a `reference_id` or a [UUID-128](https://tools.ietf.org/html/rfc4122).
```

Comment:
So because this one is a string and not an actual uuid that's where we're going to introduce another variant that requires that its a uuid (encoded as a fixed length byte array)

Reply:
:shrug:

this is what's in the DIP and pervasive at this point in time :(